coap/client.go

```
@@ -11,27 +11,39 @@ import (
 	"log"
 	"time"
 
-	"github.com/plgd-dev/go-coap/v2/message"
-	"github.com/plgd-dev/go-coap/v2/message/codes"
-	"github.com/plgd-dev/go-coap/v2/udp"
-	"github.com/plgd-dev/go-coap/v2/udp/client"
-	"github.com/plgd-dev/go-coap/v2/udp/message/pool"
+	"github.com/plgd-dev/go-coap/v3/message"
+	"github.com/plgd-dev/go-coap/v3/message/codes"
+	"github.com/plgd-dev/go-coap/v3/message/pool"
+	"github.com/plgd-dev/go-coap/v3/mux"
+	"github.com/plgd-dev/go-coap/v3/options"
+	"github.com/plgd-dev/go-coap/v3/udp"
+	"github.com/plgd-dev/go-coap/v3/udp/client"
 )
 
-var errInvalidMsgCode = errors.New("message can be GET, POST, PUT or DELETE")
+var (
+	maxRetries        uint32 = 10
+	errInvalidMsgCode        = errors.New("message can be GET, POST, PUT or DELETE")
+)
 
 // Client represents CoAP client.
 type Client struct {
-	conn *client.ClientConn
+	conn *client.Conn
 }
 
 // New returns new CoAP client connecting it to the server.
-func New(addr string) (Client, error) {
+func New(addr string, keepAlive uint64) (Client, error) {
+	if keepAlive > 0 {
+		c, err := udp.Dial(addr, options.WithKeepAlive(maxRetries, time.Duration(keepAlive)*time.Second, onInactive))
+		if err != nil {
+			log.Fatalf("Error dialing: %v", err)
```

Let's return the error and handle it on main